crates/ruff_python_formatter/src/expression/expr_bin_op.rs

```diff
@@ -27,120 +29,188 @@ pub struct FormatExprBinOp;
 impl FormatNodeRule<ExprBinOp> for FormatExprBinOp {
     fn fmt_fields(&self, item: &ExprBinOp, f: &mut PyFormatter) -> FormatResult<()> {
         let comments = f.context().comments().clone();
-
-        match Self::layout(item, f.context()) {
-            BinOpLayout::LeftString(expression) => {
```

Comment:
The `LeftString` was a "cheap" implementation of the new formatting logic that only supported implicit concatenated strings at the left of a binary expression. The new formatting supports implicit concatenated strings in arbitrary positions.